packages/template-retail-react-app/app/hooks/use-search-params.js

```diff
@@ -42,6 +42,9 @@ export const useSearchParams = (searchParams = DEFAULT_SEARCH_PARAMS) => {
 export const stringify = (searchParamsObj) => {
     let searchParamsObjCopy = {...searchParamsObj}
 
+    // Remove our copy of the original refinement value so it's not stringified.
+    delete searchParamsObjCopy._refine
```

Comment:
This is not necessary if `_refine` is a non-enumerable property (set via `Object.defineProperty(params, '_refine', { value: params.refine })`).

Reply:
I like that. But as it site it won't work without some further refactoring because the `useSearchParams` hook uses destructuring which is effected by the enumerability of the object.